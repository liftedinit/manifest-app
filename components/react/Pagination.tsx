import React, { createContext } from 'react';

export interface PaginationProps<T> {
  pageSize: number;
  defaultPage?: number;
  selectedPage?: number;
  dataset: T[];

  onChange?: (data: T[], page: number) => void;
  children?: React.ReactNode | ((data: T[], page: number) => React.ReactNode);
}

/**
 * Creates an array of page indices based on the total number of items, page
 * size, current page, and maximum number of visible pages. The array may
 * include the string '...' to represent skipped page ranges.
 *
 * @param total - The total number of items across all pages.
 * @param pageSize - The number of items per page.
 * @param [current=0] - The current page index (zero-based). Default is 0.
 * @param [max=8] - The maximum number of visible page indices. Default is 8.
 * @return An array of page indices (zero-based) and/or '...' to represent
 *         skipped pages.
 */
export function createArrayOfPageIndex(
  total: number,
  pageSize: number,
  current = 0,
  max = 8
): (number | '...')[] {
  if (max <= 4) {
    throw new Error('max cannot be less than 5');
  }
  const nbPages = Math.ceil(total / pageSize);
  current = Math.max(Math.min(nbPages - 1, current), 0);

  const toShow = Math.min(nbPages, max);
  if (nbPages <= max) {
    return [...Array(nbPages).keys()];
  } else if (current <= toShow / 2) {
    return [...Array(toShow - 2).keys(), '...', nbPages - 1];
  } else if (current >= nbPages - 1 - toShow / 2) {
    return [
      0,
      '...',
      ...Array(toShow - 2)
        .keys()
        .map(x => x + nbPages - toShow / 2 - 2), // -2 to account for 0, and last.
    ];
  } else {
    let start = current - Math.floor(toShow / 2 - 1) + 1;
    let end = current + toShow / 2 - 1;
    return [
      0,
      '...',
      ...Array(end - start)
        .keys()
        .map(x => x + start),
      '...',
      nbPages - 1,
    ];
  }
}

export function Pagination<T>({
  pageSize,
  defaultPage = 0,
  selectedPage,
  dataset,
  onChange,
  children,
}: PaginationProps<T>) {
  const [pageInner, setPageInner] = React.useState(defaultPage);
  const pages = createArrayOfPageIndex(dataset.length, pageSize, pageInner);
  const nbPages: number = Math.ceil(dataset.length / pageSize);

  React.useEffect(() => {
    if (selectedPage !== undefined) {
      setPageInner(selectedPage);
    }
  }, [selectedPage]);

  function callOnChange() {
    onChange?.(dataset.slice(pageInner * pageSize, (pageInner + 1) * pageSize), pageInner);
  }

  function previous() {
    setPageInner(page => Math.max(0, page - 1));
    callOnChange();
  }

  function next() {
    setPageInner(page => Math.min(nbPages - 1, page + 1));
    callOnChange();
  }

  function setPage(page: number) {
    setPageInner(page);
    callOnChange();
  }

  const data = dataset.slice(pageInner * pageSize, (pageInner + 1) * pageSize);

  return (
    <>
      {children instanceof Function ? (
        children(data, pageInner)
      ) : (
        <Pagination.Index.Provider value={pageInner}>
          <Pagination.Data.Provider value={data}>{children}</Pagination.Data.Provider>
        </Pagination.Index.Provider>
      )}

      <div className="flex items-center justify-end gap-2 mt-4">
        <button
          onClick={previous}
          disabled={pageInner == 0}
          aria-label="Previous page"
          className="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
        >
          ‹
        </button>

        {pages.map((p, i) => {
          if (p === '...') {
            return (
              <span key={`ellipsis-${i}`} className="text-black dark:text-white">
                ...
              </span>
            );
          } else if (p === pageInner) {
            return (
              <button
                key={`page-${p}`}
                aria-label={`Page ${p}`}
                aria-current="page"
                className="w-8 h-8 flex items-center justify-center rounded-lg transition-colors bg-[#0000001A] dark:bg-[#FFFFFF1A] text-black dark:text-white"
              >
                {p + 1}
              </button>
            );
          } else {
            return (
              <button
                key={`page-${p}`}
                onClick={() => setPage(p)}
                aria-label={`Page ${p}`}
                className="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
              >
                {p + 1}
              </button>
            );
          }
        })}

        <button
          onClick={next}
          disabled={pageInner === nbPages - 1}
          aria-label="Next page"
          className="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
        >
          ›
        </button>
      </div>
    </>
  );
}

Pagination.Index = createContext<number>(-1);
// Unfortunately, we cannot set the type of the context for Data to T, as T is
// unknown here.
Pagination.Data = createContext<any[]>([]);
