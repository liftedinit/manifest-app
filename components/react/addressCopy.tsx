import React, { useState, useEffect } from 'react';
import { FiCopy, FiCheck } from 'react-icons/fi';

export const TruncatedAddressWithCopy = ({
  address,
  slice,
  size,
}: {
  address: string;
  slice: number;
  size?: string;
}) => {
  const [copied, setCopied] = useState(false);

  useEffect(() => {
    let timer: ReturnType<typeof setTimeout>;
    if (copied) {
      timer = setTimeout(() => setCopied(false), 2000);
    }
    return () => clearTimeout(timer);
  }, [copied]);

  const handleCopy = async (e: React.MouseEvent<HTMLDivElement>) => {
    e.stopPropagation();
    e.preventDefault();
    try {
      await navigator.clipboard.writeText(address);
      setCopied(true);
    } catch (err) {
      console.error('Failed to copy: ', err);
    }
  };

  const truncatedAddress = `${address?.slice(0, slice)}...${address?.slice(-6)}`;
  const iconSize = size === 'small' ? 10 : 16;

  return (
    <span
      className="flex items-center hover:text-primary dark:hover:text-primary space-x-2 text-[#00000099] dark:text-[#FFFFFF99]"
      onClick={handleCopy}
      style={{ cursor: 'pointer' }}
    >
      <span className="truncate ">{truncatedAddress}</span>
      {copied ? <FiCheck size={iconSize} /> : <FiCopy size={iconSize} />}
    </span>
  );
};

export const AddressWithCopy = ({ address }: { address: string }) => {
  const [copied, setCopied] = useState(false);

  useEffect(() => {
    let timer: ReturnType<typeof setTimeout>;
    if (copied) {
      timer = setTimeout(() => setCopied(false), 2000);
    }
    return () => clearTimeout(timer);
  }, [copied]);

  const handleCopy = async () => {
    try {
      await navigator.clipboard.writeText(address);
      setCopied(true);
    } catch (err) {
      console.error('Failed to copy: ', err);
    }
  };

  return (
    <span
      className="flex items-center space-x-2"
      onClick={handleCopy}
      style={{ cursor: 'pointer' }}
    >
      <span className="hover:text-primary">{address}</span>
      {copied ? <FiCheck size="16" /> : <FiCopy size="16" />}
    </span>
  );
};
