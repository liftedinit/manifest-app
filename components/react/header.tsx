import env from '@/config/env';
import { WalletSection } from '../wallet';

export default function Header() {
  return (
    <header>
      <nav className="bg-white border-b border-gray-200 dark:bg-gray-800 dark:border-gray-700 px-4 fixed w-full lg:px-6 py-2 ">
        <div className="flex justify-end items-center mx-auto max-w-screen-6xl">
          {/* Log Out Button */}
          <WalletSection chainName={env.chain} />
        </div>
      </nav>
    </header>
  );
}
