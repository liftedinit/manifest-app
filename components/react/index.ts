export * from './addressCopy';
export * from './modal';
export * from './views';
export * from './settingsModal';
export * from './inputs';
export { SignModal } from './authSignerModal';
export * from './ModalDialog';
