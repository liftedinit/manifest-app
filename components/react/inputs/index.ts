export * from './AmountInput';
export * from './TextInput';
export * from './NumberInput';
export * from './TextArea';
