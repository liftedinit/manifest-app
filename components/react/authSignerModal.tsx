import { useEffect, useState } from 'react';
import { SignData } from '@cosmos-kit/web3auth';
import { TxBody, AuthInfo } from '@liftedinit/manifestjs/dist/codegen/cosmos/tx/v1beta1/tx';
import { decodePubkey } from '@cosmjs/proto-signing';
import { useWallet, useChain } from '@cosmos-kit/react';
import { getRealLogo } from '@/utils';
import { useTheme } from '@/contexts';
import env from '@/config/env';
import { ArrowRightIcon } from '../icons';
import { objectSyntax } from '../messageSyntax';
import { MsgSend } from '@liftedinit/manifestjs/dist/codegen/cosmos/bank/v1beta1/tx';
import {
  MsgCreateGroupWithPolicy,
  MsgSubmitProposal,
  MsgUpdateGroupMembers,
  MsgUpdateGroupPolicyMetadata,
  MsgUpdateGroupPolicyDecisionPolicy,
  MsgUpdateGroupMetadata,
} from '@liftedinit/manifestjs/dist/codegen/cosmos/group/v1/tx';
import {
  MsgCancelUpgrade,
  MsgSoftwareUpgrade,
} from '@liftedinit/manifestjs/dist/codegen/cosmos/upgrade/v1beta1/tx';
import { MsgSetPower } from '@liftedinit/manifestjs/dist/codegen/strangelove_ventures/poa/v1/tx';
import {
  MsgPayout,
  MsgBurnHeldBalance,
} from '@liftedinit/manifestjs/dist/codegen/liftedinit/manifest/v1/tx';
import {
  MsgSetDenomMetadata,
  MsgCreateDenom,
} from '@liftedinit/manifestjs/dist/codegen/osmosis/tokenfactory/v1beta1/tx';

type DisplayDataToSignProps = {
  data: SignData;
  address: string;
};

// Message decoder registry
const messageDecoders: Record<string, (value: Uint8Array) => any> = {
  '/cosmos.bank.v1beta1.MsgSend': (value: Uint8Array) => {
    const decoded = MsgSend.decode(value);
    return { ...decoded };
  },
  '/cosmos.group.v1.MsgCreateGroupWithPolicy': (value: Uint8Array) => {
    const decoded = MsgCreateGroupWithPolicy.decode(value);
    return { ...decoded };
  },
  '/cosmos.group.v1.MsgSubmitProposal': (value: Uint8Array) => {
    const decoded = MsgSubmitProposal.decode(value);
    return { ...decoded };
  },
  '/cosmos.group.v1.MsgUpdateGroupMetadata': (value: Uint8Array) => {
    const decoded = MsgUpdateGroupMetadata.decode(value);
    return { ...decoded };
  },
  '/cosmos.group.v1.MsgUpdateGroupPolicyMetadata': (value: Uint8Array) => {
    const decoded = MsgUpdateGroupPolicyMetadata.decode(value);
    return { ...decoded };
  },
  '/cosmos.group.v1.MsgUpdateGroupPolicyDecisionPolicy': (value: Uint8Array) => {
    const decoded = MsgUpdateGroupPolicyDecisionPolicy.decode(value);
    return { ...decoded };
  },
  '/cosmos.group.v1.MsgUpdateGroupMembers': (value: Uint8Array) => {
    const decoded = MsgUpdateGroupMembers.decode(value);
    return { ...decoded };
  },
  '/cosmos.upgrade.v1beta1.MsgCancelUpgrade': (value: Uint8Array) => {
    const decoded = MsgCancelUpgrade.decode(value);
    return { ...decoded };
  },
  '/cosmos.upgrade.v1beta1.MsgSoftwareUpgrade': (value: Uint8Array) => {
    const decoded = MsgSoftwareUpgrade.decode(value);
    return { ...decoded };
  },
  '/strangelove_ventures.poa.v1.MsgSetPower': (value: Uint8Array) => {
    const decoded = MsgSetPower.decode(value);
    return { ...decoded };
  },
  '/liftedinit.manifest.v1.MsgPayout': (value: Uint8Array) => {
    const decoded = MsgPayout.decode(value);
    return { ...decoded };
  },
  '/liftedinit.manifest.v1.MsgBurnHeldBalance': (value: Uint8Array) => {
    const decoded = MsgBurnHeldBalance.decode(value);
    return { ...decoded };
  },
  '/osmosis.tokenfactory.v1beta1.MsgSetDenomMetadata': (value: Uint8Array) => {
    const decoded = MsgSetDenomMetadata.decode(value);
    return { ...decoded };
  },
  '/osmosis.tokenfactory.v1beta1.MsgCreateDenom': (value: Uint8Array) => {
    const decoded = MsgCreateDenom.decode(value);
    return { ...decoded };
  },
};

const DisplayDataToSign = ({
  data,
  address,
  className,
  addressClassName,
  txInfoClassName,
  theme,
}: DisplayDataToSignProps & {
  className?: string;
  addressClassName?: string;
  txInfoClassName?: string;
  theme?: string;
}) => {
  const [isAddressExpanded, setIsAddressExpanded] = useState(false);
  const [isTxInfoExpanded, setIsTxInfoExpanded] = useState(false);

  const decodeBodyBytes = (bodyBytes: Uint8Array) => {
    try {
      const decodedBody = TxBody.decode(bodyBytes);
      return {
        messages: decodedBody.messages.map(msg => {
          const base64Value = Buffer.from(msg.value).toString('base64');

          try {
            // Check if we have a specific decoder for this message type
            if (messageDecoders[msg.typeUrl]) {
              return {
                typeUrl: msg.typeUrl,
                value: messageDecoders[msg.typeUrl](msg.value),
              };
            }

            // Fallback to generic base64 decoding
            const decodedValue = Buffer.from(base64Value, 'base64').toString('utf8');
            try {
              return {
                typeUrl: msg.typeUrl,
                value: JSON.parse(decodedValue),
              };
            } catch {
              return {
                typeUrl: msg.typeUrl,
                value: decodedValue,
              };
            }
          } catch (error) {
            console.error(`Failed to decode message of type ${msg.typeUrl}:`, error);
            return {
              typeUrl: msg.typeUrl,
              value: base64Value,
            };
          }
        }),
        memo: decodedBody.memo,
        timeoutHeight: decodedBody.timeoutHeight.toString(),
        extensionOptions: decodedBody.extensionOptions,
        nonCriticalExtensionOptions: decodedBody.nonCriticalExtensionOptions,
      };
    } catch (error) {
      console.error('Failed to decode bodyBytes:', error);
      return 'Failed to decode bodyBytes';
    }
  };

  const decodeAuthInfoBytes = (authInfoBytes: Uint8Array) => {
    try {
      const decodedAuthInfo = AuthInfo.decode(authInfoBytes);
      return {
        signerInfos: decodedAuthInfo.signerInfos.map(signerInfo => ({
          publicKey: signerInfo.publicKey ? decodePubkey(signerInfo.publicKey) : null,
          modeInfo: signerInfo.modeInfo,
          sequence: signerInfo.sequence.toString(),
        })),
        fee: {
          amount: decodedAuthInfo.fee?.amount,
          gasLimit: decodedAuthInfo.fee?.gasLimit.toString(),
          payer: decodedAuthInfo.fee?.payer,
          granter: decodedAuthInfo.fee?.granter,
        },
      };
    } catch (error) {
      console.error('Failed to decode authInfoBytes:', error);
      return 'Failed to decode authInfoBytes';
    }
  };

  const formatValue = (value: any, theme: string) => {
    if (value instanceof Uint8Array) {
      return Buffer.from(value).toString('base64');
    }
    if (typeof value === 'object' && value !== null) {
      if ('bodyBytes' in value && 'authInfoBytes' in value) {
        const decodedValue = {
          ...value,
          bodyBytes: decodeBodyBytes(value.bodyBytes),
          authInfoBytes: decodeAuthInfoBytes(value.authInfoBytes),
        };
        return objectSyntax(
          JSON.parse(
            JSON.stringify(decodedValue, (_, v) => (typeof v === 'bigint' ? v.toString() : v))
          ),
          theme
        );
      }
      return objectSyntax(
        JSON.parse(JSON.stringify(value, (_, v) => (typeof v === 'bigint' ? v.toString() : v))),
        theme
      );
    }
    return String(value);
  };

  return (
    <div className={className}>
      <div className="flex flex-col gap-2">
        <button
          onClick={() => setIsAddressExpanded(!isAddressExpanded)}
          className="flex items-center gap-2 text-sm font-medium"
        >
          <div className="flex items-center gap-2 flex-row justify-between">
            <span>Address</span>
            {isAddressExpanded ? (
              <ArrowRightIcon size={12} style={{ transform: 'rotate(90deg)' }} />
            ) : (
              <ArrowRightIcon size={12} style={{ transform: 'rotate(180deg)' }} />
            )}
          </div>
        </button>
        {isAddressExpanded && <pre className={addressClassName}>{address}</pre>}
      </div>
      <div className="flex flex-col gap-2">
        <button
          onClick={() => setIsTxInfoExpanded(!isTxInfoExpanded)}
          className="flex items-center gap-2 text-sm font-medium"
        >
          <div className="flex items-center gap-2 flex-row justify-between">
            <span>Tx Info</span>
            {isTxInfoExpanded ? (
              <ArrowRightIcon size={12} style={{ transform: 'rotate(90deg)' }} />
            ) : (
              <ArrowRightIcon size={12} style={{ transform: 'rotate(180deg)' }} />
            )}
          </div>
        </button>
        {isTxInfoExpanded && (
          <div className={txInfoClassName}>{formatValue(data.value, theme ?? 'light')}</div>
        )}
      </div>
    </div>
  );
};

const SignModal = ({
  visible,
  onClose,
  data,
  approve,
  reject,
}: {
  visible: boolean;
  onClose: () => void;
  data: SignData;
  approve: () => void;
  reject: () => void;
}) => {
  const wallet = useWallet();
  const { address } = useChain(env.chain);
  const walletIcon = wallet.wallet?.logo;
  const walletName = wallet.wallet?.prettyName;
  const { theme } = useTheme();

  useEffect(() => {
    const modal = document.getElementById('sign-modal') as HTMLDialogElement;
    if (visible) {
      modal.showModal();
    } else {
      modal.close();
    }
  }, [visible]);

  const walletIconString = walletIcon?.toString() ?? '';

  return (
    <dialog id="sign-modal" className="modal top-0 right-0">
      <div className="modal-box max-w-lg w-full dark:bg-[#1D192D] bg-[#FFFFFF] rounded-lg shadow-xl">
        <div className="flex justify-between items-center pb-4">
          <div className="flex items-center gap-3">
            <img
              src={getRealLogo(walletIconString, theme === 'dark')}
              alt="Wallet type logo"
              className="w-8 h-8"
            />
            <h3 className="text-xl font-semibold">{walletName?.toString()} Direct Signer</h3>
          </div>
          <button className="btn btn-sm btn-circle btn-ghost" onClick={onClose}>
            ✕
          </button>
        </div>

        <DisplayDataToSign
          data={data}
          address={address ?? ''}
          theme={theme}
          className="space-y-4"
          addressClassName="p-3 rounded-md text-sm overflow-auto h-12 dark:bg-[#E0E0FF0A] bg-[#E0E0FF0A] dark:border-[#FFFFFF33] border-[#00000033] border"
          txInfoClassName="p-3 rounded-md text-sm overflow-auto h-[32rem] dark:bg-[#E0E0FF0A] bg-[#E0E0FF0A] dark:border-[#FFFFFF33] border-[#00000033] border"
        />

        <div className="modal-action mt-6 flex justify-between gap-4">
          <button
            className="btn flex-1 rounded-[12px] focus:outline-none dark:bg-[#FFFFFF0F] bg-[#0000000A]"
            onClick={() => {
              reject();
              onClose();
            }}
          >
            Reject
          </button>
          <button
            className="btn btn-gradient flex-1 rounded-[12px]"
            onClick={() => {
              approve();
              onClose();
            }}
          >
            Approve
          </button>
        </div>
      </div>
      <form method="dialog" className="modal-backdrop">
        <button onClick={onClose}>close</button>
      </form>
    </dialog>
  );
};

export default SignModal;
