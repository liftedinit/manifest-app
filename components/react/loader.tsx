import { useEffect, useState } from 'react';

interface LoaderProps {
  progress: number;
  loaded: boolean;
  onEnter: () => void;
}

export const Loader = ({ progress, loaded, onEnter }: LoaderProps) => {
  const [showEnterButton, setShowEnterButton] = useState(false);

  useEffect(() => {
    if (loaded) {
      // Add a small delay before showing the enter button
      const timer = setTimeout(() => setShowEnterButton(true), 500);
      return () => clearTimeout(timer);
    }
  }, [loaded]);

  return (
    <div className="flex flex-col items-center">
      <progress className="progress progress-primary w-56" value={progress} max="100" />
      <div className="mt-2 text-sm text-gray-500">Loading... {Math.round(progress)}%</div>
      {showEnterButton && (
        <button
          onClick={onEnter}
          className="mt-8 px-8 py-3 bg-primary text-white rounded-md opacity-0 animate-fade-in hover:bg-primary-dark transition-colors duration-300"
        >
          Enter
        </button>
      )}
    </div>
  );
};
