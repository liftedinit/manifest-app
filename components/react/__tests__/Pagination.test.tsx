import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, test } from 'bun:test';
import { useContext } from 'react';

import { Pagination, createArrayOfPageIndex } from '@/components/react/Pagination';
import { formatComponent } from '@/tests';

describe('createArrayOfPageIndex', () => {
  test('works for small current', () => {
    expect(createArrayOfPageIndex(10, 2)).toEqual([0, 1, 2, 3, 4]);
    expect(createArrayOfPageIndex(10, 4)).toEqual([0, 1, 2]);
  });

  test('ellipsis works', () => {
    expect(createArrayOfPageIndex(20, 2, 0)).toEqual([0, 1, 2, 3, 4, 5, '...', 9]);
    expect(createArrayOfPageIndex(20, 2, 10)).toEqual([0, '...', 4, 5, 6, 7, 8, 9]);
    expect(createArrayOfPageIndex(40, 2, 8)).toEqual([0, '...', 6, 7, 8, 9, 10, '...', 19]);
    expect(createArrayOfPageIndex(40, 2, 9)).toEqual([0, '...', 7, 8, 9, 10, 11, '...', 19]);
  });
});

describe('Pagination', () => {
  test('works for small dataset', () => {
    function PageContent() {
      const i = useContext(Pagination.Index);
      const d = useContext(Pagination.Data);
      return (
        <>
          {i} {d.join()}
        </>
      );
    }

    const mockup = render(
      <Pagination pageSize={5} dataset={['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k']}>
        <Pagination.Index.Consumer>{i => i}</Pagination.Index.Consumer>
        <Pagination.Data.Consumer>{d => JSON.stringify(d)}</Pagination.Data.Consumer>
      </Pagination>
    );

    expect(formatComponent(mockup.asFragment())).toMatchSnapshot('start');
    fireEvent.click(screen.getByLabelText(/Next page/i));
    expect(formatComponent(mockup.asFragment())).toMatchSnapshot('middle');
    fireEvent.click(screen.getByLabelText(/Next page/i));
    expect(formatComponent(mockup.asFragment())).toMatchSnapshot('end');
  });
});
