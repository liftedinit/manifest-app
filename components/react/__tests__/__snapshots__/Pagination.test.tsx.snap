// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination works for small dataset: start 1`] = `
"<DocumentFragment>
  0["a","b","c","d","e"]
  <div
    class="flex items-center justify-end gap-2 mt-4"
  >
    <button
      aria-label="Previous page"
      class="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
      disabled=""
    >
      ‹
    </button>
    <button
      aria-current="page"
      aria-label="Page 0"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors bg-[#0000001A] dark:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      1
    </button>
    <button
      aria-label="Page 1"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      2
    </button>
    <button
      aria-label="Page 2"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      3
    </button>
    <button
      aria-label="Next page"
      class="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
    >
      ›
    </button>
  </div>
</DocumentFragment>"
`;

exports[`Pagination works for small dataset: middle 1`] = `
"<DocumentFragment>
  1["f","g","h","i","j"]
  <div
    class="flex items-center justify-end gap-2 mt-4"
  >
    <button
      aria-label="Previous page"
      class="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
    >
      ‹
    </button>
    <button
      aria-label="Page 0"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      1
    </button>
    <button
      aria-current="page"
      aria-label="Page 1"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors bg-[#0000001A] dark:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      2
    </button>
    <button
      aria-label="Page 2"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      3
    </button>
    <button
      aria-label="Next page"
      class="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
    >
      ›
    </button>
  </div>
</DocumentFragment>"
`;

exports[`Pagination works for small dataset: end 1`] = `
"<DocumentFragment>
  2["k"]
  <div
    class="flex items-center justify-end gap-2 mt-4"
  >
    <button
      aria-label="Previous page"
      class="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
    >
      ‹
    </button>
    <button
      aria-label="Page 0"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      1
    </button>
    <button
      aria-label="Page 1"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      2
    </button>
    <button
      aria-current="page"
      aria-label="Page 2"
      class="w-8 h-8 flex items-center justify-center rounded-lg transition-colors bg-[#0000001A] dark:bg-[#FFFFFF1A] text-black dark:text-white"
    >
      3
    </button>
    <button
      aria-label="Next page"
      class="p-2 hover:bg-[#0000001A] dark:hover:bg-[#FFFFFF1A] text-black dark:text-white rounded-lg transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
      disabled=""
    >
      ›
    </button>
  </div>
</DocumentFragment>"
`;
